Sources/FoundationEssentials/FileManager/FileOperations+Enumeration.swift

```diff
@@ -80,6 +82,7 @@ struct _Win32DirectoryContentsSequence: Sequence {
 
                 let prefixed: String =
                     prefix + name + ((slash && ffdData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY) ? #"\"# : "")
+                defer { bValid = FindNextFileW(hFind, &ffdData) }
                 return Element(fileName: name, fileNameWithPrefix: prefixed, dwFileAttributes: ffdData.dwFileAttributes)
             } while FindNextFileW(hFind, &ffdData)
             return nil
```

Comment:
Should we set `bValid = false` here so that if the `FindNextFileW` from the `while` loop fails and we return `nil` that we don't produce values after that `nil` value if called again?

Reply:
The `repeat` is to skip over the `.` and `..` entries and re-run the iteration. Perhaps we should just add a `bValid = false` before the `return nil`.

Reply:
Yeah adding that before the `return nil` would make sense to me, just to make sure that after one `next()` invocation returns `nil` that we don't accidentally try to read from `ffdData` on the following `next()` call